src/pages/_app.tsx

```diff
@@ -55,16 +58,53 @@ Router.events.on('routeChangeStart', () => {
 Router.events.on('routeChangeComplete', () => NProgress.done())
 Router.events.on('routeChangeError', () => NProgress.done())
 
-// eslint-disable-next-line import/no-default-export
-export default function App({ Component, pageProps }: AppProps) {
-  return (
-    <React.StrictMode>
-      <ThemeProvider theme={theme}>
-        <FontFix />
-        <NProgressStyles />
-        <Component {...pageProps} />
-        <ToastNotice />
-      </ThemeProvider>
-    </React.StrictMode>
+const FrontendClientBase = dynamic<FrontendClientBaseProps>(() =>
+  import('@/components/frontend-client-base').then(
+    (mod) => mod.FrontendClientBase
   )
+)
+
+// a very specific monkey patch for the next.js router usage of history api
```

Comment:
okay, no idea what happens here. 
why do we need this?

Reply:
added a comment